src/http/server/handlers/static_file_handler/cache.cr

```diff
@@ -0,0 +1,39 @@
+module HTTP::StaticFileHandler::Cache
+  HTTP_IF_MODIFIED_SINCE = "If-Modified-Since"
+  HTTP_IF_NONE_MATCH     = "If-None-Match"
+  HTTP_ETAG              = "Etag"
+  HTTP_LAST_MODIFIED     = "Last-Modified"
+
+  def self.add_headers(response_headers : HTTP::Headers, file_path : String) : Nil
+    response_headers[HTTP_ETAG] = etag(file_path)
+
+    last_modified = modification_time(file_path)
+    response_headers[HTTP_LAST_MODIFIED] = HTTP.rfc1123_date(last_modified)
+  end
+
+  def self.match?(context : HTTP::Server::Context, file_path : String) : Bool
+    # According to RFC 7232:
+    # A recipient must ignore If-Modified-Since if the request contains an If-None-Match header field
+    if if_none_match = context.request.headers[HTTP_IF_NONE_MATCH]?
```

Comment:
```cr
case
when if_none_match = context.request.headers[HTTP_IF_NONE_MATCH]?
  # ...
when if_modified_since = context.request.headers[HTTP_IF_MODIFIED_SINCE]?
  # ...
else
  # ...
end
```

Reply:
to be honest by first glance I don't see how one is better than other (and vice versa) 😅 I thought at first with crystal's implicit-object syntax there might be some more _sophisticated_ approach, maybe let's just leave it as-it?

Reply:
This is clearly a case for `if`. Yes, it could be written as `case` but that's a bit different semantically. And it's more to type. Theoretically, you could rewrite every `if ... elseif` as `case ... when`, but what would be the point?